codec/proto_codec.go

```diff
@@ -138,12 +138,8 @@ func (pc *ProtoCodec) MustUnmarshalLengthPrefixed(bz []byte, ptr ProtoMarshaler)
 // NOTE: this function must be used with a concrete type which
 // implements proto.Message. For interface please use the codec.MarshalInterfaceJSON
 func (pc *ProtoCodec) MarshalJSON(o gogoproto.Message) ([]byte, error) {
-	m, ok := o.(ProtoMarshaler)
-	if !ok {
-		return nil, fmt.Errorf("cannot protobuf JSON encode unsupported type: %T", o)
-	}
```

Comment:
want to point this section out. its unclear if we need something like this still?

Reply:
No, but if we pass in a type that implements protoreflect but isn't from gogo it could be a problem because under the hood this uses gogo proto json. But that's true of all the methods on codec pretty much.

Reply:
Should I do anything to fix?

Reply:
Not right now I think. But there will be potential errors. Not sure we can or should do anything about that.